pkg/distributor/field_detection_test.go

```diff
@@ -167,6 +167,106 @@ func Test_DetectLogLevels(t *testing.T) {
 			Value: constants.LogLevelTrace,
 		})
 	})
+
+	t.Run("detected_level with uppercase value gets normalized to lowercase", func(t *testing.T) {
+		limits, ingester := setup(true)
+		distributors, _ := prepare(t, 1, 5, limits, func(_ string) (ring_client.PoolClient, error) { return ingester, nil })
+
+		// Create a write request with detected_level in uppercase
+		writeReq := makeWriteRequestWithLabels(1, 10, []string{`{foo="bar"}`}, false, false, false)
+		writeReq.Streams[0].Entries[0].Line = `some log message`
+		writeReq.Streams[0].Entries[0].StructuredMetadata = push.LabelsAdapter{
+			{
+				Name:  constants.LevelLabel, // detected_level
+				Value: "ERROR",              // Uppercase value
+			},
+		}
+
+		_, err := distributors[0].Push(ctx, writeReq)
+		require.NoError(t, err)
+		topVal := ingester.Peek()
+		require.Equal(t, `{foo="bar"}`, topVal.Streams[0].Labels)
+
+		// Verify that detected_level is normalized to lowercase
+		sm := topVal.Streams[0].Entries[0].StructuredMetadata
+		require.Len(t, sm, 1)
+		require.Equal(t, constants.LevelLabel, sm[0].Name)
+		require.Equal(t, constants.LogLevelError, sm[0].Value) // Should be lowercase "error"
+	})
+
+	t.Run("detected_level with mixed case value gets normalized to lowercase", func(t *testing.T) {
+		// Test various mixed case values
+		testCases := []struct {
+			input    string
+			expected string
+		}{
+			{"WaRn", constants.LogLevelWarn},
+			{"InFo", constants.LogLevelInfo},
+			{"CRITICAL", constants.LogLevelCritical},
+			{"Debug", constants.LogLevelDebug},
+			{"FaTaL", constants.LogLevelFatal},
+			{"tRaCe", constants.LogLevelTrace},
+		}
+
+		for _, tc := range testCases {
+			// Create a fresh setup for each test case
+			limits, ingester := setup(true)
```

Comment:
Why is a separate setup needed for each test? Is there some state that would cause problems?

Reply:
IMO, the tests for the log level extraction does not need any of this setup. `extractLogLevel` can be tested independently. A single test case with the distributor/ingester setup to test push payloads reach this function would be enough to test the function in a wider context.

Reply:
It's far more important to cover the different cases (level in labels, structured metadata, and log entry). This is likely already done in the existing test cases.

Reply:
`Push` is the closest public method to `extractLogLevel`. The reason to test at that level was to avoid testing private methods, and to test behavior through the public API. All the other tests for this logic were already following this pattern, so it seemed best to continue to follow that pattern. The reason for the separate ingester per test case was to make assertions easier as we can always pull `topVal`. I can refactor that for efficiency though.

Reply:
these tests are a pain, as the setup (shared by lots of other distributor tests) hardcode a replication factor of 3 and setup at least 2 ingesters. when I try just using one ingester for all the test cases I find it hard to get the correct stream because some pushes go to the other ingester. fixing any of this feels like a much bigger task than this small PR warrants.